scarb/src/bin/scarb/commands/commands.rs

```diff
@@ -1,9 +1,159 @@
 use anyhow::Result;
+use serde::{Serialize, Serializer};
+use std::collections::BTreeMap;
+use std::fmt::Display;
+use std::path::PathBuf;
+use std::{env, fs};
 
+use crate::args::ScarbArgs;
 use scarb::core::Config;
+use scarb::process::is_executable;
+use scarb::ui::Message;
+use scarb::EXTERNAL_CMD_PREFIX;
+
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Serialize)]
+pub enum CommandInfo {
+    BuiltIn { about: Option<String> },
+    External { path: PathBuf },
+}
+
+impl Display for CommandInfo {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
```

Comment:
```suggestion
use std::fmt;

impl fmt::Display for CommandInfo {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
```

nitpick: let's be consistent here with rest of code